decidim-budgets/db/migrate/20200714103519_move_budgets_to_own_model.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class MoveBudgetsToOwnModel < ActiveRecord::Migration[5.2]
+  class Component < ApplicationRecord
+    self.table_name = :decidim_components
+  end
+
+  class Budget < ApplicationRecord
+    self.table_name = :decidim_budgets_budgets
+  end
+
+  class Project < ApplicationRecord
+    self.table_name = :decidim_budgets_projects
+  end
+
+  class Order < ApplicationRecord
+    self.table_name = :decidim_budgets_orders
+  end
+
+  def up
+    Project.reset_column_information
+    Order.reset_column_information
+
+    budget_components.each do |component|
+      resource = create_budget_resource_from(component)
+
+      if resource
+        add_budget_references_to_projects(resource)
+        add_budget_reference_to_orders(resource)
+      end
+    end
+
+    remove_column :decidim_budgets_projects, :decidim_component_id
+    remove_column :decidim_budgets_orders, :decidim_component_id
+  end
+
+  def down
+    add_column :decidim_budgets_projects, :decidim_component_id
+    add_column :decidim_budgets_orders, :decidim_component_id
+
+    Budget.find_each do |resource|
+      revert_budget_to_component(resource)
+      if resource
+        add_component_reference_to_projects(resource)
+        add_component_reference_to_orders(resource)
+      end
+    end
+  end
+
+  # up methods
+  def budget_components
+    @budget_components ||= Component.where(manifest_name: "budgets")
+  end
+
+  def create_budget_resource_from(component)
+    component_total_budget = (component["settings"]["global"]["total_budget"] if component["settings"]["global"].try(:key?, "total_budget"))
```

Comment:
maybe you can use the `dig` method here: `component.dig("settings", "global", "total_budget")`